### PR TITLE
Pull Request #2

### PR DESCRIPTION
**Pull request by TomKimsey**
Changed project from a demo framework application to a generic production oriented connector.

### CODE REVIEW

File: OSIsoft_Demo/src/JSONArray.java [hunk 1]
```diff
@@ -0,0 +1,891 @@
+
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/JSONArray.java]: Should we put this stuff in a subdir to separate it out from the main code?

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import com.ewon.ewonitf.EWException;
+import com.ewon.ewonitf.ScheduledActionManager;
+
+/**
+ * OSIsoftServer class
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: Thoughts on not re-using the class name in the comments here or below? Makes it easier to rename the class in the future.

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 3]
```diff
@@ -53,4 +61,29 @@ private static String buildBody(String value) {
       return jsonBody;
    }
    
+   // Reads a file and returns it as a string
+   public static String readFile(String path) {
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: Should probably document that upon any error returns an empty string. Its tough to know how, as the caller, there was an empty file though vs. an error.

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 4]
```diff
@@ -45,6 +45,74 @@ public OSIsoftServer(String ip, String login, String webID) {
...
+         items = requestResponse.getJSONArray("Items");
+         if (items.length() > 0) {
+            //tag exists
+            tag.setWebID(items.getJSONObject(0).getString("WebId"));
+         }
+         else {
+            //tag does not exist, error
+            System.out.println("Error: PI Point creation failed");
+         }
+
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: whitespace?

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 5]
```diff
@@ -45,6 +45,74 @@ public OSIsoftServer(String ip, String login, String webID) {
       postHeaders = "Authorization=Basic " + authCredentials + "&Content-Type=application/json";
    }
    
+   // Looks up and sets a tags PI Point web id.  If the tag does not exist
+   // in the data server a PI Point is created for that tag.
+   public void setTagWebId(Tag tag) throws JSONException {
+      
+      //HTTPS responses are stored in this file
+      String responseFilename = "/usr/response.txt";
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: we have to store the response into a file vs. a variable?

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 6]
```diff
@@ -45,6 +45,74 @@ public OSIsoftServer(String ip, String login, String webID) {
...
+         try {
+            ScheduledActionManager.RequestHttpX(url, "Post", postHeaders,buildNewPointBody(tag.getTagName()), "", responseFilename);
+         } catch (EWException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+         }
+         
+         //The WebID is sent back in the headers of the previous post
+         //however, there is no mechanism currently to retrieve it so 
+         //another request must be issues.
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: s/issues/issued

File: OSIsoft_Demo/src/OSIsoftServer.java [hunk 7]
```diff
@@ -45,6 +45,74 @@ public OSIsoftServer(String ip, String login, String webID) {
       postHeaders = "Authorization=Basic " + authCredentials + "&Content-Type=application/json";
    }
    
+   // Looks up and sets a tags PI Point web id.  If the tag does not exist
+   // in the data server a PI Point is created for that tag.
+   public void setTagWebId(Tag tag) throws JSONException {
+
```
Comment (cmo-hms) [inline OSIsoft_Demo/src/OSIsoftServer.java]: whitespace?

File: OSIsoft_Demo/src/OSIsoftServer.java (hunk 5 above)
Comment (TomKimsey) [reply OSIsoft_Demo/src/OSIsoftServer.java]: Unfortunately yes.

Comment (cmo-hms) [reply OSIsoft_Demo/src/OSIsoftServer.java]: A comment about how we can't retrieve the values into a variable would make it more clear imo about the current limitation of requesthttpx.

**Comment by TomKimsey** [comment]
@cmo-hms  changes made

File: OSIsoft_Demo/src/JSONArray.java (hunk 1 above)
Comment (TomKimsey) [reply OSIsoft_Demo/src/JSONArray.java]: added a commit to the end to put these files in the JSON folder

File: OSIsoft_Demo/src/OSIsoftServer.java (hunk 2 above)
Comment (TomKimsey) [reply OSIsoft_Demo/src/OSIsoftServer.java]: Removed this

File: OSIsoft_Demo/src/OSIsoftServer.java (hunk 3 above)
Comment (TomKimsey) [reply OSIsoft_Demo/src/OSIsoftServer.java]: Documented in the method

**Pull request merged**